app/src/main/java/com/github/onlynotesswent/ui/overview/FolderContent.kt

```diff
@@ -0,0 +1,236 @@
+  userViewModel.currentUser.collectAsState().value?.let {
+    folderViewModel.getSubFoldersOf(folder.value?.id!!)
+  }
+
+  val parentFolderId = folderViewModel.parentFolderId.collectAsState()
+
+  var expanded by remember { mutableStateOf(false) }
+  var showRenameDialog by remember { mutableStateOf(false) }
+  var showCreateDialog by remember { mutableStateOf(false) }
+  var updatedName by remember { mutableStateOf(folder.value!!.name) }
+  var expandedFolder by remember { mutableStateOf(false) }
+
+  Scaffold(
+      modifier = Modifier.testTag("folderContentScreen"),
+      topBar = {
+        TopAppBar(
+            colors = TopAppBarDefaults.topAppBarColors(containerColor = Color(0xFFB3E5FC)),
+            title = {
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    Spacer(modifier = Modifier.weight(2f))
+                    Text(updatedName, Modifier.testTag("folderContentTitle"))
+                    Spacer(modifier = Modifier.weight(2f))
+                  }
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = {
+                    navigationActions.navigateTo(TopLevelDestinations.OVERVIEW)
+                  }, // For now we always return to overview screen, go back does not work properly
+                  modifier = Modifier.testTag("clearButton")) {
+                    Icon(imageVector = Icons.Default.Clear, contentDescription = "Clear")
+                  }
+            },
+            actions = {
+              CustomDropDownMenu(
+                  modifier = Modifier.testTag("folderSettingsButton"),
+                  modifierItem1 = Modifier.testTag("renameFolderButton"),
+                  modifierItem2 = Modifier.testTag("deleteFolderButton"),
+                  fabIcon = {
+                    Icon(imageVector = Icons.Default.MoreVert, contentDescription = "settings")
+                  },
+                  expanded = expanded,
+                  onFabClick = { expanded = true },
+                  onDismissRequest = { expanded = false },
+                  textItem1 = { Text("Rename Folder") },
+                  textItem2 = { Text("Delete Folder") },
+                  onClickItem1 = {
+                    expanded = false
+                    showRenameDialog = true
+                  },
+                  onClickItem2 = {
+                    expanded = false
+                    folderViewModel.deleteFolderById(folder.value!!.id, folder.value!!.userId)
+                    // for now we just delete the folder directly
+                    // later on we need to figure out how to recursively delete all elements of a
+                    // folder(folders and notes)
+                    navigationActions.navigateTo(TopLevelDestinations.OVERVIEW)
+                  })
+            })
+      },
+      floatingActionButton = {
+        CustomDropDownMenu(
```

Comment:
Precisely I created CustomDropdownMenu to make it more scalable and modular. Creating another custom composable would only add overhead to the code, since it fulfills the same function as CustomDropdownMenu